src/wasm/wasm-type.cpp

```diff
@@ -370,7 +370,7 @@ bool Type::isTuple() const {
 
 bool Type::isRef() const {
   if (isBasic()) {
-    return id >= funcref && id <= i31ref;
+    return id >= funcref && id <= dataref;
```

Comment:
Maybe it would be worth defining more `static constexpr` values like `_last_basic_type` to make it easier to keep these up to date? For example, we could define `_first_reftype` and `_last_reftype`. On the other hand, this probably wouldn't scale well enough for a large number of categorizations of types.